website/_assets/js/tryFlow.js.es6.liquid

```diff
@@ -106,12 +106,25 @@ function removeChildren(node) {
 }
 
 function getAnnotations(text, callback, options, editor) {
-  Promise.resolve(editor.getOption('flow'))
-  .then(flowProxy => flowProxy.checkContent('-', text))
-  .then(errors => {
-    CodeMirror.signal(editor, 'flowErrors', errors);
+  const flow = Promise.resolve(editor.getOption('flow'));
+  const sources = Promise.all([
+    flow
+      .then(flowProxy => flowProxy.coverage('-', text))
```

Comment:
this code has to run on older versions of flow. I guess the `catch` will handle that, but swallowing errors always comes back to bite me. if it returns a JSON error for parse errors instead of throwing, then you could return a similar error if `flow.coverage == null`.

Reply:
How is this different from catch?

Reply:
not catching inside Promise.all, would kill whole sequence, checkContent wouldn't be called then

Reply:
> swallowing errors always comes back to bite me.

type-at-pos bites back, if you add `console.error` to type-at-pos catch it would complain about `re_string_match` not implemented 😬

Reply:
https://github.com/facebook/flow/issues/7939